decoder/nginx_test.go

```diff
@@ -0,0 +1,77 @@
+2022/08/18 09:22:10 [error] 844548#844548: *44926378 connect() failed (111: Connection refused) while connecting to upstream, client: 10.116.180.227, server: seller-admin-ui.name.tldn, request: "GET /swagger.json HTTP/1.1", upstream: "http://10.117.234.51:84/swagger.json", host: "seller-admin-ui.name.tldn:84"
+2022/08/18 09:29:37 [error] 844935#844935: *44934601 upstream timed out (110: Operation timed out) while connecting to upstream, client: 10.125.172.251, server: mpm-youtube-downloader-38.name.tldn, request: "POST /download HTTP/1.1", upstream: "http://10.117.246.15:84/download", host: "mpm-youtube-downloader-38.name.tldn:84"
+2022/08/18 09:36:34 [error] 845285#845285: *44941780 upstream timed out (110: Operation timed out) while connecting to upstream, client: 10.133.233.207, server: mpm-youtube-downloader-38.name.tldn, request: "POST /download HTTP/1.1", upstream: "http://10.117.246.15:84/download", host: "mpm-youtube-downloader-38.name.tldn:84"
+2022/08/18 09:38:24 [error] 1513920#1513920: *426522493 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 10.133.184.214, server: eparser.name.tldn, request: "POST /ozon.opendatacollecting.eparser.EParser/GetImageBytes HTTP/2.0", upstream: "grpc://10.117.156.164:82", host: "eparser.name.tldn:82"
+2022/08/18 09:38:24 [error] 1513920#1513920: *426522493 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 10.133.184.214, server: eparser.name.tldn, request: "POST /ozon.opendatacollecting.eparser.EParser/GetImageBytes HTTP/2.0", upstream: "grpc://10.117.160.158:82", host: "eparser.name.tldn:82"
+2022/08/18 09:38:25 [error] 1513524#1513524: *426522535 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 10.125.142.196, server: eparser.name.tldn, request: "POST /ozon.opendatacollecting.eparser.EParser/GetImageBytes HTTP/2.0", upstream: "grpc://10.117.160.158:82", host: "eparser.name.tldn:82"
+2022/08/18 09:38:25 [error] 1513524#1513524: *426522535 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 10.125.142.196, server: eparser.name.tldn, request: "POST /ozon.opendatacollecting.eparser.EParser/GetImageBytes HTTP/2.0", upstream: "grpc://10.117.173.178:82", host: "eparser.name.tldn:82"
+`
+
+	nginxInvalid = `
+invalid
+#invalid
+2022/08/18 09:38:25
+`
+)
+
+func TestDecodeNginxPass(t *testing.T) {
+	reader := strings.NewReader(nginxValid)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Bytes()
+		if len(line) == 0 {
+			continue
+		}
+		root := insaneJSON.Spawn()
+		err := DecodeNginxError(root, line)
+		if err != nil {
```

Comment:
Use `require.NoError` instead of `f.Fatal` for more homogeneity across repository.